src/models/geograph.py

```diff
@@ -586,6 +622,13 @@ def get_graph_components(
             graph components.
         """
         components: List[set] = list(nx.connected_components(graph))
-        geom = [self.df.geometry.iloc[list(comp)].unary_union for comp in components]
+        geom = [self.df.geometry.loc[comp].unary_union for comp in components]
         gdf = gpd.GeoDataFrame({"geometry": geom}, crs=self.df.crs)
         return gdf, components
+
+    def identify_node(
+        self, node_id: int, other_graph: GeoGraph, mode: str
+    ) -> List[int]:
```

Comment:
This function currently returns an ndarray, do we want to convert it to list or change the type hint?

Reply:
OK, fixed in 29beff4